signbank/dictionary/adminviews.py

```diff
@@ -304,6 +304,11 @@ def get_queryset(self):
             qs = qs.filter(
                 pk__in=pks_for_glosses_with_morphdefs_with_correct_role)
 
+        # Filter by usage
+        if 'usage' in get and get['usage'] != '':
+            vals = get.getlist('usage')
+            qs = Gloss.objects.filter(usage__id__in=vals)
```

Comment:
You need to filter `qs` here, not `Gloss.objects`

```suggestion
            qs = qs.filter(usage__id__in=vals)
```